packages/mobx-state-tree/src/core/node/identifier-cache.ts

```diff
@@ -68,23 +68,31 @@ export class IdentifierCache {
         }
     }
 
-    splitCache(node: AnyObjectNode): IdentifierCache {
-        const res = new IdentifierCache()
-        const basePath = node.path
+    splitCache(splitNode: AnyObjectNode): IdentifierCache {
+        const newCache = new IdentifierCache()
+        // The slash is added here so we only match children of the splitNode. In version 5.1.8 and
```

Comment:
**praise:** @scytacki - excellent catch + makes sense here. Thanks for the comments - this is an important fix, would be easy for us to lose context over time